### DISCUSSION TITLE
Difference between weight and control mode

### DISCUSSION THREAD

**Discussion by blacklig** [Mikubill/sd-webui-controlnet#1925] [Q&A]
Hi, can anyone explain me basically what is difference behind what weight I set for control net (can go from 0 to 2 where 1 is default) and behind "control mode" where default is "balanced" and then there is "My prompt is more important" and "ControlNet is more important"? What is actually difference there? If I put more weight on control net wont I get same result as if I check that control net is more important and vice versa?

Can be for example situation where it would be beneficial to use low weight and in the same time set that control net is more important etc?

Thanks!

**Comment by anyjf** [comment]
I am also very confused about this, please explain the process of Controlnet in a simple way for those who understand. The above parameters (their importance, weight) are compared between whom (Prompt Vs Controlnet? One Controlnet Vs Another? Other?).

**Comment by johnMinelli** [comment]
I've the same doubt... I'm trying to reproduce in diffusers the "ControlNet is more important" but I am kind of lost in the code...
From the README it seems to me that with such option you multiply the output controlnet output by the value of cfg_scale as well i.e. you scale further the strength scaling.
But 1. doing that in code don't result in the same output 2. I don't find reference of that in the source code 
https://github.com/Mikubill/sd-webui-controlnet/blob/10bd9b25f62deab9acb256301bbf3363c42645e7/scripts/hook.py#L568C20-L568C39

param.soft_injection=control_mode != external_code.ControlMode.BALANCED
param.cfg_injection=control_mode == external_code.ControlMode.CONTROL

Any hint?
